### DISCUSSION TITLE
GET parameters are no longer merged in 0.28.0

### DISCUSSION THREAD

**Discussion by yan12125** [encode/httpx#3428] [Potential Issue] [answered]
https://github.com/encode/httpx/pull/3364 changes how GET parameters are handled. For this example script:
```Python
import json
import httpx

print(json.dumps(httpx.get("https://httpbin.org/get?foo=bar", params={}).json()))
```
Results before the PR:
```sh
$ python t.py | jq .args
{
  "foo": "bar"
}
```
And after the PR:
```
$ python t.py | jq .args
{}
```
Is that an intentional change? If so, maybe documenting it in changelogs is better.

**Comment by lovelydinosaur** [comment, marked as answer]
> Is that an intentional change? If so, maybe documenting it in changelogs is better.

Yep, I think you've got the right take there. My recollection is that I noticed that the simpler refactoring would also fix a bug along the way, tho then forgot to point that out by the time I came to issue the refactoring. Unfortunately we didn't happen to have a test case covering this.

https://github.com/encode/httpx/issues/3433#issuecomment-2514327255

PRs updating the CHANGELOG there would be welcome. (Possibly also including related tests)

**Reply by eliasdorneles** [reply]
I see that the change is documented in the [changelog file](https://github.com/encode/httpx/blob/master/CHANGELOG.md#0280-28th-november-2024) but not [in the Github Release](https://github.com/encode/httpx/releases/tag/0.28.0).
A friend and I spent sometime debugging our broken tests, until we realized there was this change.

@tomchristie could we have the Github Release updated to match the contents of changelog section? :angel:
Thank you so much! :heart:

**Reply by lovelydinosaur** [reply]
Thanks @eliasdorneles, I've fixed up the release notes now. 😌
